packages/nx-maven/src/generators/library/generator.ts

```diff
@@ -310,6 +310,7 @@ async function libraryGenerator(
         options: {
           task: 'test',
         },
+        dependsOn: ['build'],
```

Comment:
Thanks for this.       
Can we just use : dependsOn: ['^build']
Because i don't need to build a lib to test it. Just build the dependencies?

Reply:
For me, the parallel problem is because maven don't handle concurrent-safe access to local Maven repository. Maybe this can fix it https://github.com/khalilou88/jnxplus/issues/700

Reply:
You can use `dependsOn: ['build']`  for you but I prefer `dependsOn: ['^build']`.
You can create some tests to check if test target contains `dependsOn: ['build']` like this one:

```
  it("should show build-tools-checkstyle config", () => {
    const libName = "build-tools-checkstyle";
    const projectJson = showProjectJson(libName);
    expect(Object.entries(projectJson.targets).length).toBe(1);
    expect(projectJson.targets.build.outputs).toEqual([
      "{projectRoot}/target",
      "{options.outputDirLocalRepo}",
    ]);
    expect(
      normalizePath(projectJson.targets.build.options.outputDirLocalRepo),
    ).toBe(`nx-maven/.m2/repository/com/example/${libName}/1.0`);
  });

```

check https://github.com/khalilou88/jnxplus-examples/blob/main/regression-tests/nx-maven-tests/src/nx-maven-tests.spec.ts 

Like this you are sur you use  `dependsOn: ['build']` as you want.

Reply:
* https://github.com/khalilou88/jnxplus/issues/700 is an interesting read, but I don't think that will fix the issue. I don't have concurrency issues in the mvn repository, but rather on the target folder of a project.

* I'm not sure if I understand correctly, do you want me to create an additional MR for the regression test? Why does it exist in a separate project instead of bundled with the jnxplus repository?

* I'd be happy to change `build` to `^build`, but wouldn't that make other people run into the same issue as we did? Setting it to `build` will work in any case without additional configuration of maven.

* I see that you have contact with nx about integrating nx-maven into nx, awesome! I've checked out the nx repository and found that if nx does not allow tasks of the same project to be run in parallel, we can get away with `^build` as well.

Reply:
My idea was just to add the necessary change to the plugin 

> * [Concurrent-safe access to local Maven repository #700](https://github.com/khalilou88/jnxplus/issues/700) is an interesting read, but I don't think that will fix the issue. I don't have concurrency issues in the mvn repository, but rather on the target folder of a project.
I didn't think about target folder

> * I'm not sure if I understand correctly, do you want me to create an additional MR for the regression test? Why does it exist in a separate project instead of bundled with the jnxplus repository?
No, just an idea for your monorepo to validate the configuration

> * I'd be happy to change `build` to `^build`, but wouldn't that make other people run into the same issue as we did? Setting it to `build` will work in any case without additional configuration of maven.
Okay will accept your change

> * I see that you have contact with nx about integrating nx-maven into nx, awesome! I've checked out the nx repository and found that if nx does not allow tasks of the same project to be run in parallel, we can get away with `^build` as well. I'm checking with the nx devs if they agree on this solution.
No, nx-maven will not be part of nx. Nx will publish a gradle plugin instead

Reply:
Thanks for your time! :) I contacted the nx team and they will look into the issue that I raised, as other java implementations will probably run into the same issue.

Reply:
@RobertAlblas Where did you raise that issue? Can't find anything in their GitHub repo. We have the same problem.

Reply:
@ToppScorer there is many issue related to this, I thought it was fixed by NX team :
https://github.com/nrwl/nx/issues/17334

Reply:
@khalilou88 But will it address this?

> I'd rather see an option to not allow parallel tasks within the same project in nx itself. **The consequence is that every test run becomes dependent on the build run.** So in the example above, it would generate build tasks for example-lib-1 and example-lib-2 as well, sadly.

respectively this?

> because we don't need to build a lib to test it.

For me the linked ticket is about something else.

Reply:
@ToppScorer I contacted them on their discord channel. I don't know if they created a github issue.